encoders/src/main/scala/scala3encoders/derivation/Helper.scala

```diff
@@ -11,22 +11,19 @@ import org.apache.spark.sql.catalyst.WalkedTypePath
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 
-// This is copied from spark  to support older versions of Spark and 3.5.0 -
+// This is copied from spark to support older versions of Spark and 3.5.0 -
 // it was part of ScalaReflection and was moved to EncoderUtils in 3.5.0
 object Helper {
   private val nullOnOverflow = !SQLConf.get.ansiEnabled
 
-  val typeBoxedJavaMapping: Map[DataType, Class[_]] = Map[DataType, Class[_]](
-    BooleanType -> classOf[java.lang.Boolean],
-    ByteType -> classOf[java.lang.Byte],
-    ShortType -> classOf[java.lang.Short],
-    IntegerType -> classOf[java.lang.Integer],
-    LongType -> classOf[java.lang.Long],
-    FloatType -> classOf[java.lang.Float],
-    DoubleType -> classOf[java.lang.Double],
-    DateType -> classOf[java.lang.Integer],
-    TimestampType -> classOf[java.lang.Long],
-    TimestampNTZType -> classOf[java.lang.Long]
+  val typeBoxedJavaMapping: Map[Class[?], Class[?]] = Map[Class[?], Class[?]](
+    classOf[Boolean] -> classOf[java.lang.Boolean],
+    classOf[Byte] -> classOf[java.lang.Byte],
+    classOf[Short] -> classOf[java.lang.Short],
+    classOf[Int] -> classOf[java.lang.Integer],
+    classOf[Long] -> classOf[java.lang.Long],
+    classOf[Float] -> classOf[java.lang.Float],
+    classOf[Double] -> classOf[java.lang.Double]
```

Comment:
Mapping only the primitive types from Scala to Java, the rest should be the same.

Reply:
OK - but basically removing the last 3 entries of the Map would have done the same I guess.

Reply:
... not sure why I did it that way initially - but yeah - this is the correct implementation, just mapping the basic types

Reply:
> basically removing the last 3 entries of the Map would have done the same I guess

Not for custom serializer. Try my test without it, it will fail.

Reply:
I am curious why.

> DateType -> classOf[java.lang.Integer]

seems correct since the size of the DataType is 4 bytes and the others is 8 bytes.

Reply:
@vincenzobaz the old implementation just wouldn't work with Option + DateTime type - I hadn't checked that explicitly - so non-primitive types are just  mapped to `WrapOption(ct.runtimeClass)`. This fix should be fine.